hypothesis-python/src/hypothesis/strategies/_internal/strategies.py

```diff
@@ -528,6 +528,12 @@ def _transform(self, element):
 
     def do_draw(self, data):
         result = self.do_filtered_draw(data)
+        if (
+            isinstance(result, SearchStrategy)
+            and not hasattr(data, "sampling_is_from_a_collection_of_strategies")
+            and all(isinstance(x, SearchStrategy) for x in self.elements)
+        ):
+            data.sampling_is_from_a_collection_of_strategies = True
```

Comment:
I insert a repr of the strategies here: 4948d085bf43fe56219eda198e877b4560abb9da
I just did this as a single `str` value rather than a list. Is there a reason to `append` here (and create a list if doesn't already exist)?